flasher.cpp

```diff
@@ -534,6 +563,32 @@ bool Flasher::SendMessage(const char *data, qint64 length, int timeout_ms)
     return CheckAck();
 }
 
+bool Flasher::ReadMessageWithCrc(const char *data, qint64 length, int timeout_ms, QByteArray& out_data)
+{
+    bool success = false;
+
+    if (serial_port_.isOpen()) {
+        serial_port_.write(data, length);
+        serial_port_.waitForReadyRead(timeout_ms);
+        QByteArray data = serial_port_.readAll();
+        QByteArray data_crc = data.right(kCrc32Size);
+
+        if(data.size() > kCrc32Size) {
+
+            uint32_t out_data_size = data.size() - kCrc32Size;
+            uint32_t crc = Deserialize32((uint8_t*)data_crc.data());
+            uint32_t calc_crc = crc::CalculateCrc32((uint8_t*)data.data(), out_data_size, false, false);
```

Comment:
Same here, reinterpret_cast

```suggestion
            uint32_t calc_crc = crc::CalculateCrc32(reinterpret_cast<uint8_t*>(data.data()), out_data_size, false, false);
```